src/hera/workflows/_mixins.py

```diff
@@ -687,13 +688,14 @@ def _get_arguments(self, **kwargs) -> List:
 
     def _get_parameter_names(self, arguments: List) -> Set[str]:
         """Returns the set of parameter names that are currently set on the mixin inheritor."""
-        parameters = [arg for arg in arguments if isinstance(arg, ModelParameter) or isinstance(arg, Parameter)]
-        return {p.name for p in parameters}
+        parameters = [arg for arg in arguments if isinstance(arg, (ModelParameter, Parameter))]
+        keys = [arg for arg in arguments if isinstance(arg, dict)]
+        return {p.name for p in parameters}.union(set(functools.reduce(lambda x, y: x + list(y.keys()), keys, [])))
```

Comment:
This essentially happens because Hera does not use `self._build_arguments()` [here](https://github.com/argoproj-labs/hera/blob/d956276f67f73a14ea72d43de494d156998b5f42/src/hera/workflows/_mixins.py#L676). When Hera uses `self.arguments` the arguments are _not_ built yet, hence it does not actually see the names of the other parameters _post_ processing from dict to `Parameter`s. An alternative solution to this PR is to call build first